src/main/kotlin/org/wycliffeassociates/otter/common/wav/WavOutputStream.kt

```diff
@@ -0,0 +1,112 @@
+package org.wycliffeassociates.otter.common.io.wav
+
+
+import java.io.BufferedOutputStream
+import java.io.Closeable
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
+import java.io.RandomAccessFile
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+
+
+/**
+ * Created by sarabiaj on 10/4/2016.
+ */
+class WavOutputStream @Throws(FileNotFoundException::class)
+@JvmOverloads constructor(
+    internal val wav: WavFile,
+    append: Boolean = false,
+    flag: BUFFERING = BUFFERING.UNBUFFERED
+) : OutputStream(), Closeable, AutoCloseable {
+
+    enum class BUFFERING {
+        BUFFERED,
+        UNBUFFERED
+    }
+
+    private val outputStream: OutputStream
+    private lateinit var bos: BufferedOutputStream
+    private var audioDataLength: Int = 0
+    private var buffered = false
```

Comment:
I agree, 
A lot of this was old java code I had tried to just get the IDE to convert to kotlin, rather than rewriting/refactoring much of it (which is what brought in a lot of some of the oddities like using === )

 Not meaning to say I didn't want to clean it up; but just giving some context lol.

Some of it should get some substantial rewriting though, like the FloatRingBuffer, which mixes too much use case specific oddities into something that is misleadingly advertised as general purpose. =\